avogadro/qtplugins/noncovalent/noncovalent.cpp

```diff
@@ -310,69 +320,80 @@ QWidget *NonCovalent::setupWidget()
 {
   QWidget *widget = new QWidget(qobject_cast<QWidget *>(this->parent()));
   QVBoxLayout *v = new QVBoxLayout;
-
-  // angle tolerance
-  QDoubleSpinBox *angle_spin = new QDoubleSpinBox;
-  angle_spin->setRange(0.0, 180.0);
-  angle_spin->setSingleStep(1.0);
-  angle_spin->setDecimals(0);
-  angle_spin->setSuffix(tr(" °"));
-  angle_spin->setValue(m_angleToleranceDegrees);
-  QObject::connect(angle_spin, QOverload<double>::of(&QDoubleSpinBox::valueChanged), this, &NonCovalent::setAngleTolerance);
+  QTabWidget *tabs = new QTabWidget;
   
-  // maximum distance
-  QDoubleSpinBox *distance_spin = new QDoubleSpinBox;
-  distance_spin->setRange(1.0, 10.0);
-  distance_spin->setSingleStep(0.1);
-  distance_spin->setDecimals(1);
-  distance_spin->setSuffix(tr(" Å"));
-  distance_spin->setValue(m_angleToleranceDegrees);
-  QObject::connect(distance_spin, QOverload<double>::of(&QDoubleSpinBox::valueChanged), this, &NonCovalent::setMaximumDistance);
+  for (Index i = 0; i < 3; i++) {
+  	// angle tolerance
+		QDoubleSpinBox *angle_spin = new QDoubleSpinBox;
+		angle_spin->setRange(0.0, 180.0);
+		angle_spin->setSingleStep(1.0);
+		angle_spin->setDecimals(0);
+		angle_spin->setSuffix(tr(" °"));
+		angle_spin->setValue(m_angleTolerancesDegrees[i]);
+		QObject::connect(angle_spin, QOverload<double>::of(&QDoubleSpinBox::valueChanged), this,
+			[this, i](float width){ return setAngleTolerance(width, i); }
+		);
+		
+		// maximum distance
+		QDoubleSpinBox *distance_spin = new QDoubleSpinBox;
+		distance_spin->setRange(1.0, 10.0);
+		distance_spin->setSingleStep(0.1);
+		distance_spin->setDecimals(1);
+		distance_spin->setSuffix(tr(" Å"));
+		distance_spin->setValue(m_maximumDistances[i]);
+		QObject::connect(distance_spin, QOverload<double>::of(&QDoubleSpinBox::valueChanged), this, 
+			[this, i](float width){ return setMaximumDistance(width, i); }
+		);
   
-  // line width
-  QDoubleSpinBox* lineWidth_spin = new QDoubleSpinBox;
-  lineWidth_spin->setRange(1.0, 10.0);
-  lineWidth_spin->setSingleStep(0.5);
-  lineWidth_spin->setDecimals(1);
-  lineWidth_spin->setValue(m_lineWidths[0]);
-  QObject::connect(lineWidth_spin, QOverload<double>::of(&QDoubleSpinBox::valueChanged), this, &NonCovalent::setLineWidth);
-
-  QFormLayout *form = new QFormLayout;
-  form->addRow(QObject::tr("Angle tolerance:"), angle_spin);
-  form->addRow(QObject::tr("Maximum distance:"), distance_spin);
-  form->addRow(QObject::tr("Line width:"), lineWidth_spin);
-  v->addLayout(form);
-
+  	// line width
+		QDoubleSpinBox* lineWidth_spin = new QDoubleSpinBox;
+		lineWidth_spin->setRange(1.0, 10.0);
+		lineWidth_spin->setSingleStep(0.5);
+		lineWidth_spin->setDecimals(1);
+		lineWidth_spin->setValue(m_lineWidth);
```

Comment:
If you put the line width in the tab, it looks like there are 3 different line width options.